internal/rukpak/preflights/crdupgradesafety/crdupgradesafety.go

```diff
@@ -32,17 +32,18 @@ type Preflight struct {
 
 func NewPreflight(crdCli apiextensionsv1client.CustomResourceDefinitionInterface, opts ...Option) *Preflight {
 	changeValidations := []kappcus.ChangeValidation{
```

Comment:
How much further would we need to go to fully drop the kapp dependency?

Reply:
Not 100% sure, but level of effort depends on whether we want to re-implement in our own way such that we don't need to adhere to Apache 2.0 license guidelines or if we would want to copy-paste the code we need including maintaining and adhering to the Apache 2.0 license guidelines

Reply:
That being said, I would say totally dropping the kapp dependency should be seen outside the scope of this PR. While this PR reduces our use of the carvel provided validators, the main reason for this PR is to address some minor bugs in the output when the checks fail. My approach to doing this just happens to reduce the carvel validator use because I felt it would be easier for us to maintain long-term as opposed to contributing the changes all the way upstream.

Reply:
Are the check implementations copy/pasted from kapp or new implementations? I ask because:
- If they are copy/pasted:
    - I don't need to review them as carefully,
    - BUT, we need some sort of attribution?
- If they are not copy/pasted,
    - We should review thoroughly
    - We don't need attribution

Reply:
They are not copy/paste-ed but *inspired* by the carvel implementation (which I helped write so hard to not be inspired by), but I think the implementation is changed significantly enough that it doesn't require any attribution